contract-tests/hook.rb

```diff
@@ -0,0 +1,68 @@
+require 'ldclient-rb'
+
+class Hook
+  include LaunchDarkly::Interfaces::Hooks::Hook
+
+  #
+  # @param name [String]
+  # @param callback_uri [String]
+  # @param data [Hash]
+  #
+  def initialize(name, callback_uri, data)
+    @metadata = LaunchDarkly::Interfaces::Hooks::Metadata.new(name)
+    @callback_uri = callback_uri
+    @data = data
+    @context_filter = LaunchDarkly::Impl::ContextFilter.new(false, [])
+  end
+
+  def metadata
+    @metadata
+  end
+
+  #
+  # @param hook_context [LaunchDarkly::Interfaces::Hooks::EvaluationContext]
+  # @param data [Hash]
+  #
+  def before_evaluation(hook_context, data)
+    payload = {
+      evaluationHookContext: {
+        flagKey: hook_context.key,
+        context: @context_filter.filter(hook_context.context),
```

Comment:
Why the filtered context?

Reply:
The SDK has no other way of converting a context into a hash other than attribute redaction. We haven't needed it yet, and unless the tests require some special handling of private attributes (which seems odd), I figured I could get away with this.